packages/utils/src/geo.ts

```
@@ -143,3 +169,514 @@ export function parseGeoPoint(point: GeoPointInput, throwInvalid = true): GeoPoi
+    const queryFeature = makeGeoFeatureOrThrow(queryGeo);
+
+    return (input: unknown): boolean => {
+        const inputGeoEntity = toGeoJSON(input);
+        if (!inputGeoEntity) return false;
+
+        const inputFeature = makeGeoFeature(inputGeoEntity);
+        if (!inputFeature) return false;
+
+        return disjoint(inputFeature, queryFeature);
+    };
+}
+
+/** Returns true if both geo entities have no overlap */
+export function geoDisjoint(firstGeoEntity: GeoInput, secondGeoEntity: GeoInput):boolean {
+    return geoDisjointFP(firstGeoEntity)(secondGeoEntity);
+}
+
+const esTypeMap = {
+    [ESGeoShapeType.Point]: GeoShapeType.Point,
+    [ESGeoShapeType.MultiPolygon]: GeoShapeType.MultiPolygon,
+    [ESGeoShapeType.Polygon]: GeoShapeType.Polygon,
+} as const;
+
+/** Only able to convert geo-points to either a geo-json point or a simple polygon.
+ * There is no current support for creating polygon with holes or multi-polygon
+ * as of right now. geoJSON input is made sure to be properly formatted for its type value
+ */
+export function toGeoJSON(input: unknown): GeoShape|undefined {
+    if (isGeoJSON(input)) {
+        const { type: inputType } = input;
+        const type = esTypeMap[inputType] ? esTypeMap[inputType] : inputType;
+        return { ...input, type };
+    }
+
+    if (isGeoPoint(input)) {
+        const coordinates = makeCoordinatesFromGeoPoint(parseGeoPoint(input as GeoPointInput));
+        return {
+            type: GeoShapeType.Point,
+            coordinates
+        };
+    }
+
+    if (Array.isArray(input)) {
+        try {
+            const points = input.map((point) => makeCoordinatesFromGeoPoint(
+                parseGeoPoint(point as GeoPointInput)
+            ));
+
+            const coordinates = validateListCoords(points);
+
+            return {
+                type: GeoShapeType.Polygon,
+                coordinates
+            };
+        } catch (_err) {
+            // ignore here
```

Why are we ignoring the error here?

---

this is not non-throwing version, we use toGeoJSONOrThrow for strict parsing